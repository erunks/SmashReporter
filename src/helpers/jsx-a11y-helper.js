export function buttonize(action) {
  return {
    role: 'button',
    onClick: action,
    onKeyDown: (event) => {
      // Handles Enter/Return pressed
      if (event.keycode === 13) {
        action(event);
      }
    }
  };
}

export default { buttonize };
