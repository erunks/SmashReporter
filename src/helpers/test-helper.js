import React from 'react';
import { render } from '@testing-library/react';
import { mount, shallow } from 'enzyme';
import { IKContext } from 'imagekitio-react';
import { RecoilRoot } from 'recoil';

const wrapper = (component, options = {
  initializeState: () => {},
}) => {
  const initialStateMethod = options.initializeState;

  return (
    <IKContext
      publicKey="public_DfZPshOEvWdxzvxS8HGNWPJCo70="
      urlEndpoint="https://ik.imagekit.io/ozcerk4wii"
    >
      <RecoilRoot initializeState={ initialStateMethod }>
        { component }
      </RecoilRoot>
    </IKContext>
  );
};

const mountWrapper = (component, options) => mount(wrapper(component, options));
const shallowWrapper = (component, options) => shallow(wrapper(component, options));
const wrappedRender = (component, options = {}) => {
  const initialStateMethod = options && options.initializeState;

  const Wrapper = ({ children }) => (
    <IKContext
      publicKey="public_DfZPshOEvWdxzvxS8HGNWPJCo70="
      urlEndpoint="https://ik.imagekit.io/ozcerk4wii"
    >
      <RecoilRoot initializeState={ initialStateMethod }>
        { children }
      </RecoilRoot>
    </IKContext>
  );

  return render(component, { wrapper: Wrapper, ...options });
};


export * from '@testing-library/react';

export {
  mountWrapper,
  shallowWrapper,
  wrappedRender
};
