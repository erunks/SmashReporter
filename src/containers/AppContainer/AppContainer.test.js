import React from 'react';
import { mountWrapper } from 'helpers/test-helper';
import App from '.';

describe('<App>', () => {
  const container = mountWrapper(<App/>);

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
