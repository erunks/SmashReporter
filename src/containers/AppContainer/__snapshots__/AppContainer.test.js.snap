// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> should match the snapshot 1`] = `"<div class=\\"FighterContainer\\"><div class=\\"FighterChoice faceLeft\\" role=\\"button\\"><img src=\\"https://www.smashbros.com/assets_v2/img/fighter/thumb_v/random.png\\" alt=\\"fighter: random\\"><div class=\\"helpText\\"><p>Click to Select Fighter</p></div></div><div class=\\"FighterSelection\\">0</div><div class=\\"FighterChoice \\" role=\\"button\\"><img src=\\"https://www.smashbros.com/assets_v2/img/fighter/thumb_v/random.png\\" alt=\\"fighter: random\\"><div class=\\"helpText\\"><p>Click to Select Fighter</p></div></div></div>"`;
